FeatureFileGenerator/feature_file_compiler.py

```diff
@@ -7,76 +7,117 @@
-        return line
-    html_tags = re.compile(r"<[^>]*>")
-    line = html_tags.sub("", line)
-    return unescape(line.replace("&nbsp;", " "))
-
-
-def feature_file_printer(feature_as_list, name):
-    feature_name = name + ".feature"
-    base_path = Path(__file__).parent
-    file_path = (base_path / "../behave/features/" / feature_name).resolve()
-
-    with open(file_path, "w+") as f:
-        f.write("Feature: " + name + "\n\r")
-        for line in feature_as_list:
-            f.write(line + "\n")
-
-
-def feature_file_maker(entity_type, tp_entity_id):
-    test_cases = entity_types[entity_type](tp_entity_id)
-    return test_cases
-
-
-def test_stepper(test_cases, feature_as_list):
-    test_cases_unloaded = []
-    for test_case in test_cases:
-        test_cases_unloaded.append(json.loads(test_case))
-
-    for test_case in test_cases_unloaded:
-        feature_as_list.append(test_case["Name"])
-        for test_step in test_case["TestSteps"]["Items"]:
-            description = test_step["Description"]
-            if "Examples" in description:
-                table = description.split("|")
-                for line in table:
-                    if strip_html(line) != "":
-                        if "Examples:" not in line:
-                            line = "|" + line + "|"
-                        feature_as_list.append(strip_html(line))
-            else:
-                feature_as_list.append(strip_html(description))
-        feature_as_list.append("")
-    return feature_as_list
+        ff = FeatureFileCompiler(tp_entity_id, args)
+        ff.get_entity_type()
+        ff.get_entity_name()
+        ff.get_all_test_cases()
+        ff.test_stepper()
+        ff.feature_file_writer()
+
+
+class FeatureFileCompiler():
+    def __init__(self, tp_id, args):
+        self.tp_id = tp_id
+        self.args = args
+        self.feature = []
+
+    def get_entity_type(self):
+        self.entity_type = requester.entity_type_getter(self.tp_id)
+
+    def get_all_test_cases(self):
+        self.test_cases = entity_types[self.entity_type](self.tp_id)
+
+    def get_entity_name(self):
+        if(self.entity_type == "TestCase"):
+            self.entity_name = "Test Case " + self.tp_id
+        else:
+            self.entity_name = requester.entity_name_getter(self.tp_id)
+    
+    def test_stepper(self):
+        for test_case in self.test_cases:
+            self.tagger(test_case)
+            if("Scenario: " not in test_case["Name"] or "Scenario Outline: " not in test_case["Name"]): 
+                self.feature.append("Scenario: " + test_case["Name"])
+            for test_step in test_case["TestSteps"]["Items"]:
+                description = test_step["Description"]
+                result = test_step["Result"]
+                if("Examples" in description):
+                    table = description.split("|")
+                    for line in table:
+                        if FeatureFileCompiler.strip_html(line) != "":
```

Comment:
This whole function could do with being broken down, it's a bit too big.

Reply:
How dare you! Yeah, agreed :) I had taken out the adding of tags bit, will go back later and split this down, luckily there are distinct "chunks"